### PR TITLE
Fix random segfault when loading cache

### PR DESCRIPTION
**Pull request by loukiano**
This diff mirrors https://github.com/jg513/enif_protobuf/pull/35.

The current stack_ensure_all() casts spot->node->fields to ep_field_t to iterate through a given node's fields, but these nodes can also be enums with fields of type ep_enum_field_t which is significantly smaller. This mismatch leads to quickly going past the end of every enum node's fields array, and field->o_type starts to index into other memory.

This usually doesn't lead to segfaults because the allocated memory for the nif state is so large that these accesses rarely go past it, so the program just silently loops through all the enums without doing anything other than access random parts of the state. This is also why the test I've added creates such a large enum, and why the issue https://github.com/jg513/enif_protobuf/issues/31 is both hard to reproduce and only happening on larger, more involved projects.

This diff fixes the issue by simply skipping enums in stack_ensure_all(), as they have to contain integer fields so there's no need to check for nested messages or maps or anything.

### CODE REVIEW

File: c_src/ep_node.c [hunk 1]
```diff
@@ -724,6 +724,12 @@ stack_ensure_all(ErlNifEnv *env, ep_cache_t *cache)
                     spot--;
                     continue;
                 }
+                
+                // skip over enums -- can't be nested
+                if (spot->node->n_type == node_enum) {
```
Comment (kennedyjosh) [inline c_src/ep_node.c:729]: I'm a bit confused here – you want to skip over enums but the condition seems to be true when it **is** an enum? Shouldn't this be the other way around?

Comment (kennedyjosh) [reply c_src/ep_node.c:729]: oh, nevermind – I think I see. this avoids going into the following for loop if it is an enum. Right?

**Comment by kennedyjosh** [comment]
The description here mentions that you added a test with a large enum, but this doesn't appear to be the case. Did you forget to add it?

File: c_src/ep_node.c (hunk 1 above)
Comment (loukiano) [reply c_src/ep_node.c:729]: yep!

**Review by Thomas-Neill — APPROVED** [review]
good stuff! will be pulling this into #21 from the upstream.

**Review by kennedyjosh — APPROVED** [review]
Great work!

**Comment by loukiano** [comment]
Ah, am realizing this may have been merged too early -- it was accepted into the main fork so I feel confident that it's good. Happy to revert if necessary.

**Pull request merged**